src/collective/taxonomy/jsonimpl.py

```diff
@@ -137,10 +141,14 @@ def __call__(self):
         )
 
     def generate_data_for_taxonomy(self, parsed_data, language, path=PATH_SEPARATOR):
+        data = json.loads(self.request.get("BODY", ""))
+        taxonomy = queryUtility(ITaxonomy, name=data["taxonomy"])
+        default_language = taxonomy.default_language
```

Comment:
This code looks very brittle...

Imagine:

```python
self.request.get("BODY", "")
```
evaluates to `""`.
Then you will have:

```python
>>> json.loads("")
Traceback (most recent call last):
...
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

but every line that follow can easily fail if the request body is not exactly what you expect.